apollo-router/src/plugins/telemetry/config_new/router/selectors.rs

```diff
@@ -875,4 +897,26 @@ mod test {
             "No Content".into()
         );
     }
+
+    #[test]
+    fn router_response_body() {
+        let selector = RouterSelector::ResponseBody {
+            response_body: true,
+        };
+        let context = crate::context::Context::new();
+        context.extensions().with_lock(|ext| {
+            ext.insert(RouterResponseBodyExtensionType(
+                r#"{"data": {"data": "res"}}"#.to_string(),
+            ));
+        });
```

Comment:
Instead of doing this manually, let's use the `stash_the_body_in_extensions` you created